src/pages/Applications/Filter/index.tsx

```diff
@@ -0,0 +1,101 @@
+import { FormValues as FV } from "./types";
+import { ApplicationsQueryParams } from "types/aws";
+import Icon, { DrawerIcon } from "components/Icon";
+import { dateToFormFormat } from "components/form";
+import { cleanObject } from "helpers/cleanObject";
+import Form from "./Form";
+import { schema } from "./schema";
+
+type Props = {
+  classes?: string;
+  setParams: React.Dispatch<React.SetStateAction<ApplicationsQueryParams>>;
+  isDisabled: boolean;
+};
+
+export default function Filter({ setParams, classes = "", isDisabled }: Props) {
+  const buttonRef = useRef<HTMLButtonElement | null>(null);
+
+  const methods = useForm<FV>({
+    mode: "onChange",
+    reValidateMode: "onChange",
+    resolver: yupResolver(schema),
+    defaultValues: {
+      //set default value so empty can be tagged as invalid
+      startDate: dateToFormFormat(getYearAgo()),
+      endDate: dateToFormFormat(new Date()),
+      hqCountry: { name: "", flag: "", code: "" },
+      status: { label: "Under Review", value: "Under Review" },
+    },
+  });
+
+  const { handleSubmit, reset } = methods;
+
+  function ISOdate(date: string, end?: boolean) {
+    if (!date) return "";
+
+    const output = new Date(date);
+    //include up to the last second of the day
+    if (end) output.setHours(23, 59, 59);
+
+    return output.toISOString();
+  }
+
+  async function submit(data: FV) {
+    setParams(
+      cleanObject({
+        regDateStart: ISOdate(data.startDate),
+        regDateEnd: ISOdate(data.endDate, true),
+        regStatus: data.status.value,
+        hqCountry: data.hqCountry.name,
+      })
+    );
+    buttonRef.current?.click();
+  }
+
+  const onReset: FormEventHandler<HTMLFormElement> = () => {
+    reset();
+    setParams({
+      limit: 10,
+      regStatus: "Under Review",
+    });
+    buttonRef.current?.click();
+  };
+  return (
+    <Popover className={`${classes} flex relative items-center`}>
+      <Popover.Button
+        ref={buttonRef}
+        disabled={isDisabled}
+        className="w-full lg:w-[22.3rem] flex justify-center items-center p-3 rounded bg-orange text-white lg:dark:text-gray lg:text-gray-d1 lg:bg-white lg:dark:bg-blue-d6 lg:justify-between disabled:bg-gray lg:disabled:bg-gray-l3 lg:dark:disabled:bg-bluegray-d1 lg:border lg:border-prim"
+      >
+        {({ open }) => (
+          <>
+            <Icon className="lg:hidden" type="Filter" size={20} />
+            <div className="uppercase font-semibold text-[0.9375rem] ">
+              Filter
+            </div>
+            <DrawerIcon isOpen={open} className="hidden lg:inline" size={21} />
+          </>
+        )}
+      </Popover.Button>
+
+      <FormProvider {...methods}>
+        <Form
+          submit={handleSubmit(submit)}
+          onReset={onReset}
+          classes="max-lg:fixed max-lg:inset-x-0 max-lg:top-0 lg:mt-1 absolute top-full z-20"
+        />
+      </FormProvider>
+    </Popover>
+  );
+}
+
+function getYearAgo() {
```

Comment:
Two things about this function: 
1. Could move to `components/form/helper.ts` since it's pretty generic and a better fit to live there rather than here.
2. We should consider a generic function that deducts N weeks or days (like `getWeeksAgo(weeks: number) => Date`) since fetching 1 year back worth of data for Under Review by default is unrealistic as they will only be open for review around ~3-7 days. User can always adjust the date range wider if needed.

Reply:
> We should consider a generic function that deducts N weeks or days (like getWeeksAgo(weeks: number)

created `weeksAgo` function  https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2484/commits/fac8549f2c75e84ef5f1d9634acbcd4174d091de

> Could move to components/form/helper.ts 

I placed it in `/helpers` in general as operating on dates isn't necessarily related to form